pageserver/src/tenant.rs

```diff
@@ -2268,6 +2287,31 @@ impl Tenant {
 
         size::gather_inputs(self, logical_sizes_at_once, &mut *shared_cache).await
     }
+
+    /// Calculate synthetic tenant size
+    /// This is periodically called by background worker
+    ///
+    #[instrument(skip_all, fields(tenant_id=%self.tenant_id))]
+    pub async fn calculate_synthetic_size(&self) -> anyhow::Result<u64> {
+        let inputs = self.gather_size_inputs().await?;
+
+        let size = inputs.calculate()?;
+
+        info!(
+            "calculate_synthetic_size for tenant {} size: {}",
+            self.tenant_id, size
+        );
+
+        self.cached_synthetic_tenant_size
+            .store(size, Ordering::Relaxed);
+
+        Ok(size)
+    }
```

Comment:
I think this could be moved to the background worker, since there's not much to do either way.